datacube_stats/incremental_stats.py

```diff
@@ -40,6 +40,50 @@ def _proc(*x):
     return _proc
 
 
+def compose_proc(input_transform, proc, output_transform=None):
+    """
+    Apply input transform before feeding data into an accumulator, optionally
+    apply output transform on the result also.
+
+      out_proc(x) == proc( input_transform(x) )
+      out_proc()  == output_transform(proc())
+
+    """
+
+    if output_transform is None:
+        output_transform = lambda x: x
+
+    def _proc(*x):
+        if len(x) == 0:
```

Comment:
Just
```python
if not x:
    ...
```
should work fine too! (It's just that I find it more readable 😄, just saying!)